code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -158,6 +162,32 @@ GLOBAL_LIST_INIT(metal_recipes, list ( \
 	user.visible_message(span_suicide("[user] begins whacking [user.p_them()]self over the head with \the [src]! It looks like [user.p_theyre()] trying to commit suicide!"))
 	return BRUTELOSS
 
+/obj/item/stack/sheet/iron/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
+	if(istype(target, /turf/open))
```

Comment:
I think this being instant can be open to abuse, could you add a slight delay to it?

Reply:
nvm I missed the part with the do_after